systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java

```diff
@@ -174,6 +183,162 @@ void testJSONFormatLogging() {
+
+        LOGGER.info("Changing rootLogger level to INFO with inline logging");
+        InlineLogging ilInfo = new InlineLogging();
+        ilInfo.setLoggers(Collections.singletonMap("rootLogger.level", "INFO"));
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(ilInfo);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(ilInfo);
+        });
+
+        LOGGER.info("The EO shouldn't roll - verifying pod stability");
+        DeploymentUtils.waitForNoRollingUpdate(eoDeploymentName, eoPods);
+
+        LOGGER.info("Waiting for log4j2.properties will contain desired settings");
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=INFO")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=INFO")
+        );
+
+        LOGGER.info("Waiting {} ms for INFO log will appear", RECONCILIATION_INTERVAL * 2);
+        //wait some time if TO and UO will log something
+        Thread.sleep(RECONCILIATION_INTERVAL * 2);
+
+        LOGGER.info("Asserting if log will contain some records");
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "user-operator", "5m"), is(not(emptyString())));
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "topic-operator", "5m"), is(not(emptyString())));
+
+        LOGGER.info("Setting log level of TO and UO to OFF - no records should appear in log");
+        // change inline logging
+
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(ilOff);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(ilOff);
+        });
+
+        LOGGER.info("The EO shouldn't roll - verifying pod stability");
+        DeploymentUtils.waitForNoRollingUpdate(eoDeploymentName, eoPods);
+
+        LOGGER.info("Waiting for log4j2.properties will contain desired settings");
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=OFF")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=OFF")
+        );
+
+        LOGGER.info("Waiting {} ms for INFO log will disappear", RECONCILIATION_INTERVAL * 2);
+        //wait some time if TO and UO will log something
+        Thread.sleep(RECONCILIATION_INTERVAL * 2);
```

Comment:
Quite a lot of waiting here. Is all needed? 
* You first wait for _some unknown_ time for rolling update to not happen. How long is that waiting?
* Then you wait for the ConfigMap reload
* Than you wait 2*Reconciliation interval for the logging to ... reload?

I wonder if this can be done faster. For example maybe you can check that the pod did not roll just after the other waits instead of waiting for it on the beginning?

This applies all over this test - maybe it is really what is needed ... but I wanna raise it just in case you did not considered it.

Reply:
The first thing is 50s (it is the same as testing pod stability).
I do wait for `2*Reconciliation` because it is the worst-case time after which is the logging really set to the new value. Reconciliation in tests is 30s so in the worst case the `log4j2.properties` file is regenerated 30s after the change is applied. Then, there is another 30s `monitorInterval` after which the configuration is reloaded.

Reply:
Just because today the intervals in the tests are the same you should not count on it for the future. So maybe it is better to use the timeout based on your calculation and explain it in comments than use 2*Reconciliation which might change and be different in the future (e.g. the default reconciliation in CO is 2 minutes and not 30 seconds, so it might not be something to rely on if your logic is based on 30 seconds comming from something else).